tests/jsunit/event_test.js

```diff
@@ -573,3 +671,75 @@ function helper_addMoveEventParent(events, block, parent) {
   block.setParent(parent);
   events[events.length-1].recordNew();
 }
+
+function test_events_newblock_newvar() {
+  eventTest_setUpWithMockBlocks();
+
+  Blockly.Events.fire = temporary_fireEvent;
+  temporary_fireEvent.firedEvents_ = [];
+  // Expect three calls to genUid: one to set the block's ID, one for the event
+  // group's id, and one for the variable's ID.
+  setUpMockMethod(mockControl_, Blockly.utils, 'genUid', null, ['1', '2', '3']);
+  try {
+    var block = workspace.newBlock('field_variable_test_block');
+
+    var firedEvents = workspace.undoStack_;
+    // Expect two events: varCreate and block create.
+    assertEquals(2, firedEvents.length);
+
+    var event0 = firedEvents[0];
+    var event1 = firedEvents[1];
+    assertEquals('var_create', event0.type);
+    assertEquals('create', event1.type);
+
+    // Expect the events to have the same group ID.
+    assertEquals(event0.group, event1.group);
+
+    // Expect the group ID to be the result of the second call to genUid.
+    assertEquals('2', event0.group);
+
+    // Expect the workspace to have a variable with ID '3'.
+    assertNotNull(workspace.getVariableById('3'));
+    assertEquals('3', event0.varId);
+  } finally {
+    eventTest_tearDownWithMockBlocks();
+    Blockly.Events.fire = savedFireFunc;
```

Comment:
Is there not a way to set this before every test function, instead of relying on individual tests to remember to set it it back?

Other test files use a convention of setup/teardown pair of methods that would be appropriate for this.

The whole setup/try/finally/teardown process could be simplified via a wrapper function builder:

```js
var test_func_whatever = buildTest(function() { ... });  // Or arrow func

function buildTest(testImp) {
  return function() {
    setup();
    try { testImpl(); }
    finally { teardown(); }
    // Maybe return any result from testImpl()
  }
}
```


All of this can be a later PR.